### ISSUE TITLE
Edge label projection not updated with control point calculation

### ISSUE THREAD

**Issue by ktei** [cytoscape/cytoscape.js#887]
@maxkfranz  I found this issue:
In screenshot 1, we have 2 nodes with 2 edges between them, edge autorotate turned on. However, if you hide any of the edge, the label on the remaining edge will be misplaced (screenshot 2). This happens regardless if it's 2 edges or X edges.

Screenshot 1:
![capture1](https://cloud.githubusercontent.com/assets/2195250/6859561/4de9da9c-d46e-11e4-89f3-e7ae12b5a737.PNG)
Screenshot 2:
![capture2](https://cloud.githubusercontent.com/assets/2195250/6859562/52c74932-d46e-11e4-8641-7b4c909c94fc.PNG)

Currently I just go to `renderer.canvas.drawing-label-text.js` and uncomment `this.recalculateEdgeLabelProjection( edge )` regardless. But I don't think this is a proper fix. Can you suggest a proper fix? Many thanks.

To reproduce this, copy the html below and open console to run `window.cy.edges()[0].hide()`.
```html
<!DOCTYPE html>
<html>
<head>
    <style>
        body {
            font: 14px helvetica neue, helvetica, arial, sans-serif;
        }

        #cy {
            height: 100%;
            width: 100%;
            position: absolute;
            left: 0;
            top: 0;
        }

        #eat {
            position: absolute;
            left: 1em;
            top: 1em;
            font-size: 1em;
            z-index: -1;
            color: #c88;
        }
    </style>
    <meta charset=utf-8 />
    <title>Cytoscape.js initialisation</title>
    <script src="http://ajax.googleapis.com/ajax/libs/jquery/1/jquery.min.js"></script>
    <script src="~/Vendor/cytoscape.js/build/cytoscape.js"></script>
</head>
<body>
    <div id="cy"></div>
    <script>
        $(function () { // on dom ready

            // photos from flickr with creative commons license

            window.cy = cytoscape({
                container: document.getElementById('cy'),

                style: cytoscape.stylesheet()
                  .selector('node')
                    .css({
                        'height': 80,
                        'width': 80,
                        'background-fit': 'cover',
                        'border-color': '#000',
                        'border-width': 3,
                        'border-opacity': 0.5,
                        'text-valign': 'bottom',
                        'text-wrap': 'wrap',
                        'content': 'test',
                    })
                  .selector('.eating')
                    .css({
                        'border-color': 'red'
                    })
                  .selector('.eater')
                    .css({
                        'border-width': 9
                    })
                  .selector('edge')
                    .css({
                        'color': '#333',
                        'width': 6,
                        'content': 'animal',
                        'target-arrow-shape': 'triangle',
                        'line-color': '#ffaaaa',
                        'target-arrow-color': '#ffaaaa',
                        'edge-text-rotation': 'autorotate',
                        'text-background-color': '#ffaaaa',
                        'text-border-color': 'red'
                    })
                  .selector('edge:selected')
                    .css({
                        'text-background-color': 'red'
                    })
                  .selector('#bird')
                    .css({
                        'background-image': 'https://farm8.staticflickr.com/7272/7633179468_3e19e45a0c_b.jpg'
                    })
                  .selector('#cat')
                    .css({
                        'background-image': 'https://farm2.staticflickr.com/1261/1413379559_412a540d29_b.jpg'
                    })
                  .selector('#ladybug')
                    .css({
                        'background-image': 'https://farm4.staticflickr.com/3063/2751740612_af11fb090b_b.jpg'
                    })
                .selector('#aphid')
                    .css({
                        'background-image': 'https://farm9.staticflickr.com/8316/8003798443_32d01257c8_b.jpg'
                    })
                .selector('#rose')
                    .css({
                        'background-image': 'https://farm6.staticflickr.com/5109/5817854163_eaccd688f5_b.jpg'
                    })
                .selector('#grasshopper')
                    .css({
                        'background-image': 'https://farm7.staticflickr.com/6098/6224655456_f4c3c98589_b.jpg'
                    })
                .selector('#plant')
                    .css({
                        'background-image': 'https://farm1.staticflickr.com/231/524893064_f49a4d1d10_z.jpg'
                    })
                .selector('#wheat')
                    .css({
                        'background-image': 'https://farm3.staticflickr.com/2660/3715569167_7e978e8319_b.jpg'
                    }),

                elements: {
                    nodes: [
                      { data: { id: 'cat' } },
                      { data: { id: 'bird' } },
                      { data: { id: 'ladybug' } },
                      { data: { id: 'aphid' } },
                      { data: { id: 'rose' } },
                      { data: { id: 'grasshopper' } },
                      { data: { id: 'plant' } },
                      { data: { id: 'wheat' } }
                    ],
                    edges: [
                      { data: { source: 'cat', target: 'bird' } },
                      { data: { source: 'bird', target: 'cat' } },
                      //{ data: { source: 'bird', target: 'ladybug' } },
                      //{ data: { source: 'bird', target: 'grasshopper' } },
                      //{ data: { source: 'grasshopper', target: 'plant' } },
                      //{ data: { source: 'grasshopper', target: 'wheat' } },
                      //{ data: { source: 'ladybug', target: 'aphid' } },
                      //{ data: { source: 'aphid', target: 'rose' } }
                    ]
                },

                layout: {
                    name: 'cose',
                    directed: true,
                    padding: 10
                }
            }); // cy init
        });
    </script>
</body>
</html>
```
